### PR TITLE
Update the interfaces documentation : EOperationStatus (included) -> IContractData (included)

### PR DESCRIPTION
**Pull request by windushka**
Aims to solve  #147

### CODE REVIEW

File: src/interfaces/IBlockcliqueBlockBySlot.ts [hunk 1]
```diff
@@ -19,6 +32,13 @@ export interface IEndorsementInfo {
   id: string;
 }
 
+/**
+ * Represents a block in a blockclique by slot.
+ * @remarks
+ * This interface is used to track the block information in a blockclique,
+ * including the block header, signature, creator's public key, creator's
+ * address, block ID, and operations.
```
Comment (Elli610) [inline src/interfaces/IBlockcliqueBlockBySlot.ts:40]: For now I do not have the knowledge to review this. I'll learn about blockclique and edit this comment later today.

File: src/interfaces/ICallData.ts [hunk 2]
```diff
@@ -1,14 +1,22 @@
+/**
+ * Represents the information for a smart contract call.
+ * @remarks
+ * This interface is used to track the call data information, including the
```
Comment (Elli610) [inline src/interfaces/ICallData.ts:4]:
```suggestion
 * This interface is used to track the calldata information, including the
```

File: src/interfaces/IClient.ts [hunk 3]
```diff
@@ -5,6 +5,13 @@ import { IPublicApiClient } from './IPublicApiClient';
 import { ISmartContractsClient } from './ISmartContractsClient';
 import { IWalletClient } from './IWalletClient';
 
+/**
+ * Represents the main client interface for interacting with the blockchain.
```
Comment (Elli610) [inline src/interfaces/IClient.ts:9]:
```suggestion
 * Represents the client object.
```

File: src/interfaces/IClient.ts [hunk 4]
```diff
@@ -5,6 +5,13 @@ import { IPublicApiClient } from './IPublicApiClient';
...
+ * @remarks
+ * This interface is used to access various APIs, including private and public
+ * APIs, wallet, and smart contracts. It also provides methods for setting
```
Comment (Elli610) [inline src/interfaces/IClient.ts:12]: This is not clear

File: src/interfaces/IClientConfig.ts [hunk 5]
```diff
@@ -1,5 +1,16 @@
 import { IProvider } from './IProvider';
 
+/**
+ * Represents the configuration for the main client interface.
```
Comment (Elli610) [inline src/interfaces/IClientConfig.ts:4]:
```suggestion
 * This interface is used to configure the client 
```

File: src/interfaces/IClique.ts [hunk 6]
```diff
@@ -1,3 +1,11 @@
+/**
+ * Represents a clique of blocks in the blockchain.
+ * @remarks
+ * This interface is used to track a block clique, including the block IDs,
+ * the fitness of the clique, and whether it is a valid block clique. Fitness
+ * is a measure of the quality of a block clique, with a higher fitness value
+ * indicating a better block clique.
```
Comment (Elli610) [inline src/interfaces/IClique.ts:7]:
```suggestion
 * - This interface is used to track a block clique, including the block IDs,
 * the fitness of the clique, and whether it is a valid block clique. 
 * - Fitness is a measure of the quality of a block clique, with a higher fitness value
 * indicating a better block clique.
```

File: src/interfaces/IContractData.ts [hunk 7]
```diff
@@ -1,14 +1,22 @@
+/**
+ * Represents the information for a smart contract.
+ * @remarks
+ * This interface is used to track the smart contract information, including the
+ * storage fee, maximum gas, contract data (as text or bytecode), contract address,
+ * and an optional datastore for the smart contract's operational storage data.
+ *
+ * @see fee of type `bigint` represents the storage fee for the smart contract.
+ * @see maxGas of type `bigint` represents the maximum gas that can be consumed by the smart contract.
```
Comment (Elli610) [inline src/interfaces/IContractData.ts:9]: is this the max gas fees that can be consumed during the deployment or while calling it ?

File: src/interfaces/IContractData.ts [hunk 8]
```diff
@@ -1,14 +1,22 @@
...
+ * @see contractDataText of type `string | undefined` represents the contract's data as text (optional).
```
Comment (Elli610) [inline src/interfaces/IContractData.ts:10]:
```suggestion
 * @see contractDataText of type `string | undefined` represents the contract's data as string (optional).
```

**Review by Elli610 — CHANGES_REQUESTED** [review]

**Review by Elli610 — APPROVED** [review]

**Review by Ben-Rey — APPROVED** [review]

**Pull request merged**